internal/unionstore/union_store.go

```diff
@@ -270,6 +249,8 @@ type MemBuffer interface {
 	FlushWait() error
 	// GetMetrics returns the metrics related to flushing
 	GetMetrics() Metrics
+	// GetSnapshot returns a snapshot of the MemBuffer.
+	GetSnapshot() MemBufferSnapshot
```

Comment:
Better to comment this is the snapshot suppose to be used and it is thread safe. 

The commonents of `SnapshotGetter()` should also be updated to clarity the differences bettween `SnapshotGetter` and the new `GetSnapshot`, when to use which.